src/components/CheckBox/CheckBox.style.ts

```diff
@@ -22,76 +22,98 @@ export const checkboxWrapperStyle = () => (theme: Theme): SerializedStyles => cs
     border-radius: 100%;
     transition: all 0.2s;
     content: ' ';
-    width: ${rem(50)};
-    height: ${rem(50)};
+    width: ${rem(48)};
+    height: ${rem(48)};
     position: absolute;
   }
   &:hover:before {
-    background: ${transparentize(0.7, theme.utils.getColor('lightGray', 400))};
+    background: rgba(0, 0, 0, 0.05);
   }
 `;
 
-export const checkboxStyle = ({ intermediate, checked }: Props) => (
-  theme: Theme
-): SerializedStyles => css`
-  background: ${checked
-    ? intermediate
-      ? theme.utils.getColor('darkGray', 400)
-      : theme.utils.getColor('branded1', 300, 'normal')
-    : theme.utils.getColor('lightGray', 300)};
-  border: 0;
-  border-radius: ${rem(2)};
-  width: ${rem(26)};
-  height: ${rem(26)};
-
-  position: absolute;
-  opacity: 0; // hide it
-
-  & + label {
-    position: relative;
-    cursor: pointer;
-    padding: 0;
-  }
+const getBackgroundColor = ({ intermediate, checked, filled, theme }: Props & { theme: Theme }) => {
+  let background = '';
```

Comment:
can i suggest remove the `let` and turn all the `background =` to `return {the color required`.

This way will be enforced to split all the if elses to a more understandable way

Reply:
eg

```
if (checked && !intermediate && filled) {
    return `background: ${theme.utils.getColor('lightGray', 700)}`;
  }

  return !filled ? background : `background: ${theme.utils.getColor('lightGray', 400)}`;
```

Reply:
done